pkg/generator/caches.go

```diff
@@ -45,20 +45,28 @@ type Caches struct {
 	listeners           []*v2.Listener
 	statusVirtualHost   *route.VirtualHost
 	logger              *zap.SugaredLogger
+	ingressesToSync     map[string]struct{}
 }
 
-func NewCaches(logger *zap.SugaredLogger, kubernetesClient kubeclient.Interface, extAuthz bool) (*Caches, error) {
+func NewCaches(logger *zap.SugaredLogger, kubernetesClient kubeclient.Interface, extAuthz bool, ingressesToSync map[string]struct{}) (*Caches, error) {
```

Comment:
This moves some complexity to the callers. Now they need to know how to build the keys that represent an ingress, which should be internal to this module. Shouldn't this method receive references to ingresses instead?

Reply:
true, check the latest changes.